gapic/src/main/com/google/gapid/views/VirtualTreeViewer.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (C) 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.gapid.views;
+
+import org.eclipse.jface.viewers.IBaseLabelProvider;
+import org.eclipse.jface.viewers.IContentProvider;
+import org.eclipse.jface.viewers.TreeViewer;
+import org.eclipse.swt.graphics.Rectangle;
+import org.eclipse.swt.widgets.Tree;
+import org.eclipse.swt.widgets.TreeItem;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * A {@link TreeViewer} that notifies the bound {@link IContentProvider} and
+ * {@link IBaseLabelProvider} of time visibility changes if they also implement
+ * the {@link VisibilityListener} interface.
+ */
+public class VirtualTreeViewer extends TreeViewer {
```

Comment:
In SWT world virtual has a specific meaning unrelated to this (see SWT.VIRTUAL). VisibilityTrackingTreeViewer is long, but since it's only ever created through the factory methods, seems ok.